### PR TITLE
POC Bold

### PR DESCRIPTION
**Pull request by FrancoisGe**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/core/commands/command_palette.js [hunk 1]
```diff
@@ -60,7 +60,58 @@ function commandsWithinCategory(categoryName, categories) {
...
+    let index = -1;
+    let nextIndex = 0;
+    const list_index = [];
+    while (nextIndex !== -1) {
+        nextIndex = lowerCaseName.indexOf(lowerCaseSearchValue, index + 1);
+        index = nextIndex;
+        if (nextIndex !== -1) {
+            list_index.push(nextIndex);
+        }
+    }
```
Comment (fbarbe00) [inline addons/web/static/src/core/commands/command_palette.js:79]: Why use two variables `index` and `nextIndex`?

```js
let index = -1;
while (true) {
    index = lowerCaseName.indexOf(lowerCaseSearchValue, index + 1);
    if (index !== -1) {
        list_index.push(index);
    }
    else {
        break;
    }
}
```
Wouldn't this work as well?

File: addons/web/static/src/core/commands/command_palette.js [hunk 2]
```diff
@@ -60,7 +60,58 @@ function commandsWithinCategory(categoryName, categories) {
...
+                isBold: true,
+            });
+            begin = index + searchValueSize;
+        }
+        if (begin < name.length) {
+            result.push({
+                name: name.substring(begin, name.length),
+                isBold: false,
+            });
+        }
```
Comment (fbarbe00) [inline addons/web/static/src/core/commands/command_palette.js:102]: Can't this loop be merged with the previous one?
```js
let index = -1;
let begin = 0;
let results = [];
while (true) {
    index = lowerCaseName.indexOf(lowerCaseSearchValue, index + 1);
    if (index !== -1) {
        result.push({
                    name: name.substring(begin, index),
                    isBold: false,
                });
        result.push({
                name: name.substring(index, index + searchValueSize),
                isBold: true,
            });
        begin = index + searchValueSize;
    }
    else {
        result.push({
                name: name.substring(begin, name.length),
                isBold: false,
            });
        break;
    }
}
```

File: addons/web/static/src/core/commands/command_palette.js [hunk 3]
```diff
@@ -59,6 +60,11 @@ function commandsWithinCategory(categoryName, categories) {
     };
 }
 
+function boldCommandName(name, searchValue) {
```
Comment (aab-odoo) [inline addons/web/static/src/core/commands/command_palette.js:63]: the function name should be adapted

**Review by aab-odoo — APPROVED** [review]
Much simpler indeed :-)

**Review by aab-odoo — APPROVED** [review]

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**